app/src/editor/canvas/index.jsx

```diff
@@ -47,9 +47,19 @@ function setUpdated(canvas) {
     return updated
 }
 
+function isEditable(canvas, runController) {
+    const { isActive } = runController
+    const canEdit = !isActive && !canvas.adhoc
+    return canEdit
```

Comment:
I suggest adding this derived state to the runController, but also include the permissions check.